types/jest/index.d.ts

```diff
@@ -198,7 +197,9 @@ declare namespace jest {
      *   spy.mockRestore();
      * });
      */
-    function spyOn<T extends {}, M extends keyof T>(object: T, method: M, accessType?: 'get' | 'set'): SpyInstance<T[M]>;
+    function spyOn<T extends {}, M extends keyof T>(object: T, method: M, accessType: 'get'): SpyInstance<T[M], []>;
+    function spyOn<T extends {}, M extends keyof T>(object: T, method: M, accessType: 'set'): SpyInstance<void, [T[M]]>;
```

Comment:
Since `T[M]` is the function type here, should this be `SpyInstance<void, ExtractArgs<T[M]>>`?

Reply:
No, TypeScript considers a setter/getter to be a property, so the type of the property is used here. It is not a function.

Reply:
`extends {}` is already the default; should be `extends object`.